ast/builtins.go

```diff
@@ -284,6 +284,21 @@ var DefaultBuiltins = [...]*Builtin{
 // built-in definitions.
 var BuiltinMap map[string]*Builtin
 
+// IgnoreDuringPartialEval is a set of built-in functions that should not be
+// evaluated during partial evaluation. These functions are not partially
+// evaluated because they are not pure.
```

Comment:
Let's add "Deprecated: ..." explanation here. https://go.dev/blog/godoc

Reply:
I have replaced the original comment with the deprecation notice.

Reply:
Amusingly, marking that item with the `Deprecated: ...` comment causes our own linters to shoot the PR down. I'll go add a manual linter ignore there. :laughing:

Reply:
Yeah that's what's been keeping us back from a few others. But really, going _this_ route instead of having non-conforming Deprecated-comments is better.